tesseract/tesseract_planning/tesseract_motion_planners/src/ompl/profile/ompl_default_plan_profile.cpp

```diff
@@ -98,15 +145,80 @@ void OMPLDefaultPlanProfile::apply(OMPLProblem& prob,
       goal_state[i] = joint_waypoint[i];
 
     prob.simple_setup->setGoalState(goal_state);
+  }
+}
 
-    // Setup state checking functionality
-    ompl::base::StateValidityCheckerPtr svc_without_collision = processStateValidator(prob, env, prob.manip_fwd_kin);
+void OMPLDefaultPlanProfile::applyStartStates(OMPLProblem& prob,
+                                              const Eigen::Isometry3d& cartesian_waypoint,
+                                              const PlanInstruction& parent_instruction,
+                                              const std::vector<std::string>& /*active_links*/,
+                                              int /*index*/)
+{
+  const auto dof = prob.manip_fwd_kin->numJoints();
 
-    // Setup motion validation (i.e. collision checking)
-    processMotionValidator(svc_without_collision, prob, env, prob.manip_fwd_kin);
+  // Check if the waypoint is not relative to the manipulator base coordinate system and at tool0
+  Eigen::Isometry3d world_to_waypoint = cartesian_waypoint;
+  if (!parent_instruction.getWorkingFrame().empty())
+    world_to_waypoint = prob.env_state->link_transforms.at(parent_instruction.getWorkingFrame()) * cartesian_waypoint;
 
-    // make sure the planners run until the time limit, and get the best possible solution
-    processOptimizationObjective(prob);
+  Eigen::Isometry3d world_to_base_link = prob.env_state->link_transforms.at(prob.manip_inv_kin->getBaseLinkName());
+  Eigen::Isometry3d manip_baselink_to_waypoint = world_to_base_link.inverse() * world_to_waypoint;
+  Eigen::Isometry3d manip_baselink_to_tool0 = manip_baselink_to_waypoint * parent_instruction.getTCP().inverse();
+
+  if (prob.state_space == OMPLProblemStateSpace::REAL_STATE_SPACE)
+  {
+    /** @todo Need to add descartes pose sampler to ompl profile */
+    Eigen::VectorXd joint_solutions;
+
+    /** @todo Need to also provide the seed instruction to use here */
+    prob.manip_inv_kin->calcInvKin(joint_solutions, manip_baselink_to_tool0, Eigen::VectorXd::Zero(dof));
+    long num_solutions = joint_solutions.size() / dof;
+    bool found_start_state = false;
+    for (long i = 0; i < num_solutions; ++i)
+    {
+      auto solution = joint_solutions.middleRows(i * dof, dof);
+      // Get descrete contact manager for testing provided start and end position
+      // This is required because collision checking happens in motion validators now
+      // instead of the isValid function to avoid unnecessary collision checks.
+      if (!checkStateInCollision(prob, solution))
```

Comment:
I'd rather we not change it here to avoid conflicts with other branches, but when we add the descartes sampler we should break this out into a utility (with it's own unit tests) to be used in multiple planners. I have similar code in a branch where I was starting to put together the joint interpolator planner, and it will likely come up in other places.

Reply:
Agree.